lightning/src/onion_message/messenger.rs

```diff
@@ -243,6 +243,66 @@ impl OnionMessageRecipient {
 	}
 }
 
+/// A struct handling response to an [`OnionMessage`]
+pub struct Responder<'a, OMH: OnionMessageHandler, T: OnionMessageContents> {
+	messenger: &'a OMH,
```

Comment:
Instead of expanding `OnionMessageHandler`'s definition, could we use a `Fn` or similar trait?